app/models/pipeline_run_stage.rb

```diff
@@ -122,27 +120,36 @@ def due_for_aegea_check?
     rand < 0.1
   end
 
-  def update_job_status
-    if !id || !started? || succeeded? || failed?
-      # This is called only from PipelineRun.update_status and only when the guard above is false.
-      Airbrake.notify("Invalid precondition for PipelineRunStage.update_job_status #{id} #{job_id} #{job_status}.")
-      return
-    end
+  def monitor_results
```

Comment:
I think we want to move monitor_results out of the pipeline_run_stages. I actually think you can just implement it in pipeline_runs with hardcoded expected results for now.

Reply:
ok!